### PR TITLE
Catch up the spring-cloud-azure-starter-keyvault-certificates in 4.1.0

### LINKED ISSUE #28238: [FEATURE REQ] GA the spring-cloud-azure-starter-keyvault-certificates
**Is your feature request related to a problem? Please describe.**
The Spring Cloud Azure 4.0 has been GAed, but with one artifact left from the first 4.0.0, the `spring-cloud-azure-starter-keyvault-certificates`. We should GA this starter in version 4.1.0, only with the feature of autoconfiguring the Key Vault certificate clients. 

**Describe the solution you'd like**
- Upgrade the starter's version to 4.1.0-beta.1.
- Delete the source code from the starter's src folder.
- Check whether the POM file is correct and succinct.

**Describe alternatives you've considered**
N/A

**Additional context**
The KV certificate starter from the 3.x contains the feature of supporting TLS, this feature will not be delivered in the 4.1.0 version, and we will use https://github.com/Azure/azure-sdk-for-java/issues/28243 to track the feature of supporting TLS, mTLS for an HTTP server/client with KV certificates. 

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Description Added
- [x] Expected solution specified

### PR DESCRIPTION
**Pull request by saragluna**
Resolves #28238.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-starter-keyvault-certificates/pom.xml [hunk 1]
```diff
@@ -86,45 +86,14 @@
...
-      <version>5.3.15</version> <!-- {x-version-update;org.springframework:spring-web;external_dependency} -->
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.apache.httpcomponents</groupId>
-      <artifactId>httpclient</artifactId>
-      <version>4.5.13</version> <!-- {x-version-update;org.apache.httpcomponents:httpclient;external_dependency} -->
-      <scope>test</scope>
+      <artifactId>azure-security-keyvault-certificates</artifactId>
+      <version>4.3.0</version> <!-- {x-version-update;com.azure:azure-security-keyvault-certificates;dependency} -->
```
Comment (moarychan) [inline sdk/spring/spring-cloud-azure-starter-keyvault-certificates/pom.xml:96]: The current certificates version is not aligned with the version of the SDK BOM.

Comment (saragluna) [reply sdk/spring/spring-cloud-azure-starter-keyvault-certificates/pom.xml:96]: The SDK BOM will release another version this month.

File: sdk/spring/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## 4.1.0-beta.1 (Unreleased)
 
 ### Features Added
+- GA the `spring-cloud-azure-starter-keyvault-certificates`. 
 
 ### Breaking Changes
```
Comment (moarychan) [inline sdk/spring/CHANGELOG.md:9]: Shall we add some description for these changes here?

Comment (saragluna) [reply sdk/spring/CHANGELOG.md:9]: Any suggestions on which aspect should we add for?

Comment (moarychan) [reply sdk/spring/CHANGELOG.md:9]: They have the same library name with different functions, I recommend we can add some description to tell uses the difference, it's a new starter.

Comment (saragluna) [reply sdk/spring/CHANGELOG.md:9]: It's called azure-spring-boot-starter-keyvault-certificates in the 3.x

Comment (saragluna) [reply sdk/spring/CHANGELOG.md:9]: But we can add the feature it supports.

**Review by stliu — APPROVED** [review]
LGTM

**Review by chenrujun — COMMENTED** [review]
LGTM.

**Pull request merged**